### PR TITLE
[PVR] Addon API: Add PVR_PROPERTIES.iEpgMaxDays, PVR_RECORDING.channelType

### PR DESCRIPTION
**Pull request by ksooo**
This PR introduces a PVR addon API change. It will get merged immediately after #9290 to keep number of api bumps for Krypton small.

1) Epg max days (number of days the user wants Kodi to hold epg event data) now will be transferred along with PVR_PROPERTIES to Addon_Create function. Thus, the pvr addon knows the correct value immediately when it gets created. Before, the value was transferred some time after the addon got created, making it impossible for the addon to initialize it's backend connection with the correct epg max days value without deferring connecting to the backend until epg max days values got set by Kodi, making a true asynchronous backend connection setup impossible.

2) Introduce PVR_RECORDING.channelType. This way, addons can flag recordings as TV or radio recordings making it possible to distinguish in kodi between these types, for instance to provide separate listings of tv and radio recordings, implement separate next recording widgets for radio and tv, .... It must be up to the addons/backend to supply this information as it is not always possible to obtain the type of a given channel at runtime, because the channel the recoding has been done may no longer be available at the time the channel type is needed. Kodi core implementation will be done in a followup PR.

Due to this API change, all PVR addons need to be recompiled, no other changes are required,  unless async epg data transfer including addon-side filtering of transferred epg data according to epg max days value shall be implemented.

Second commit fixes crash due to a multithreading issue with CGUIEPGGridContainer that occured while testing the API change.

### CODE REVIEW

**Comment by ksooo** [comment]
@Jalle19 mind taking a look

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h [hunk 1]
```diff
@@ -237,6 +237,7 @@ extern "C" {
   {
     const char* strUserPath;           /*!< @brief path to the user profile */
     const char* strClientPath;         /*!< @brief path to this add-on */
+    int iEpgMaxDays;                   /*!< @brief if > EPG_TIMEFRAME_UNLIMITED, in async epg mode, deliver only events in the range from 'end time > now' to 'start time < now + iEpgMaxDays. EPG_TIMEFRAME_UNLIMITED, notify all events. */
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: Can you use this way?
```cpp

    /// if > EPG_TIMEFRAME_UNLIMITED, in async epg mode, deliver only events in
    /// the range from 'end time > now' to 'start time < now + iEpgMaxDays.
    /// EPG_TIMEFRAME_UNLIMITED, notify all events.
    int iEpgMaxDays;
```
Is for two reasons, the first is to use the soon comming new add-on documentation and the other, `@brief` gives problems on doxygen related to structure values and text not used with them.

**Comment by AlwinEsch** [comment]
Need also a increase of `XBMC_PVR_API_VERSION`

The `XBMC_PVR_MIN_API_VERSION` is OK and can be left so.

File: xbmc/pvr/addons/PVRClient.cpp [hunk 2]
```diff
@@ -195,9 +196,6 @@ ADDON_STATUS CPVRClient::Create(int iClientId)
   catch (std::exception &e) { LogException(e, __FUNCTION__); }
 
   m_bReadyToUse = bReadyToUse;
-
-  SetEPGTimeFrame(CSettings::GetInstance().GetInt(CSettings::SETTING_EPG_DAYSTODISPLAY));
```
Comment (AlwinEsch) [inline xbmc/pvr/addons/PVRClient.cpp]: If it becomes not used anymore can be the function `PVR_ERROR SetEPGTimeFrame(int iDays);` also removed, need the only a MIN_API_VERSION increase.

**Comment by ksooo** [comment]
This pr will merged together with #9290 to avoid another api bump within a short timeframe. 9290 does the api bump.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/addons/PVRClient.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/pvr/addons/PVRClient.cpp]: It is still used. https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/PVRManager.cpp#L181

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h (hunk 1 above)
Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: I would prefer to change documentation syntax for the whole file in one PR and not to have different styles in one file at one point in time.

File: xbmc/epg/GUIEPGGridContainer.h [hunk 3]
```diff
@@ -65,9 +65,8 @@ namespace EPG
     virtual std::string GetDescription() const;
     const int GetNumChannels()   { return m_channels; };
     virtual int GetSelectedItem() const;
-    const int GetSelectedChannel() const;
     CFileItemPtr GetSelectedChannelItem() const;
-    PVR::CPVRChannelPtr GetChannel(int iIndex);
+    PVR::CPVRChannelPtr GetSelectedChannel();
```
Comment (Jalle19) [inline xbmc/epg/GUIEPGGridContainer.h]: Method should be const

Comment (ksooo) [inline xbmc/epg/GUIEPGGridContainer.h]: Yeah but not in this PR. ;-) Will ake care of such things (there are many of those in this class. e.g. "const int&" parameters) in a dedicated cleanup pr.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**